File: android/digia-ui/src/main/java/com/digia/engage/Digia.kt
package com.digia.engage

import android.content.Context
import com.digia.engage.internal.DigiaInstance

object Digia {
    fun initialize(context: Context, config: DigiaConfig) {
        DigiaInstance.initialize(context = context, config = config)
    }

    fun register(plugin: DigiaCEPPlugin) {
        DigiaInstance.register(plugin)
    }

    fun setCurrentScreen(name: String) {
        DigiaInstance.setCurrentScreen(name)
    }
}

File: iOS/Sources/DigiaEngage/api/digia.swift
import Foundation

@MainActor
public enum Digia {
    public static func initialize(_ config: DigiaConfig) {
        DigiaRuntime.shared.initialize(config)
    }

    public static func register(_ plugin: DigiaCEPPlugin) {
        DigiaRuntime.shared.register(plugin)
    }

    public static func setCurrentScreen(_ name: String) {
        DigiaRuntime.shared.setCurrentScreen(name)
    }

    public static func registerFontFactory(_ factory: DUIFontFactory) {
        DigiaRuntime.shared.registerFontFactory(factory)
    }

    @discardableResult
    public static func onMessage(
        _ name: String,
        listener: @escaping @Sendable (ScopeValue?) -> Void
    ) -> UUID {
        DigiaRuntime.shared.addMessageListener(name: name, listener: listener)
    }

    public static func removeMessageListener(_ name: String, token: UUID) {
        DigiaRuntime.shared.removeMessageListener(name: name, token: token)
    }
}
